_posts/2023-01-25-snowflake-snowpro-core-cert.md

```diff
@@ -0,0 +1,244 @@
+- Recommended compressed size of data files for optimal bulk data loads is 100 to 250 MB
+
+### Retention specific notes
+
+- Snowflake keeps the batch load history (from Stage) using COPY statement for 64 days.
+- Snowpipe load history is 14 days
+- Maximum retention period for permanent tables, schemas and databases is 90 days in Enterprise edition or higher
+- In Enterprise edition or higher For transient databases, schemas, and tables, the retention period default is set to 1 day but can be changed to 0. The same is also true for temporary tables
+- Cache query result retention period is 24 hours. However, each time the persisted result for a query is reused, Snowflake resets the 24-hour retention period for the result up to a maximum of 31 days from the date and time that the query was first executed. After 31 days, the result is purged, and the next time the query is submitted, a new result is generated and persisted.
+- Snowflake Query history page allows you to view the details of all the queries executed in the last 14 days
+- All Snowflake-managed keys are automatically rotated by Snowflake when they are more than 30 days old
+
+### Scaling policies on multi-cluster warehouses
+
+- There are two different scaling policies, one is the Standard policy, and one is the Economy policy.
+- There are two ways to set up a multi-cluster warehouse: in maximized mode, or auto-scaling mode. With maximized mode, you simply set your minimum equal to your maximum, and those values are something greater than one.
+
+### Secure data sharing
+
+Secure Data Sharing enables sharing selected objects in a database in your account with other Snowflake accounts. The following Snowflake database objects can be shared:
+
+Tables
+External tables
+Secure views
+Secure materialized views
+Secure UDFs
+
+Snowflake enables the sharing of databases through shares created by data providers and “imported” by data consumers.
+
+### Replication and cloning
+
+Temporary tables, stages, tasks, streams, pipes, and external tables are not currently supported for replication.
+
+Databases and Schemas can be cloned. External Table and Internal (Snowflake) stages do not get cloned.
+
+Database and share replication are available in all editions, including the Standard edition. Replication of all other objects is only available for Business Critical Edition (or higher).
+
+### Snowflake sampling methods
+
+SYSTEM | BLOCK sampling is often faster than BERNOULLI | ROW sampling. Also, BERNOULLI | ROW method is good for Smaller Tables, and SYSTEM | BLOCK method is for Larger Tables.
+
+### Snowflake architecture
+
+Snowflake's unique architecture consists of three key layers:
+ • Database Storage
+ • Query Processing
+ • Cloud Services
+
+## Some sample questions
+
+- **What are javascript stored procedure delimiters?**
+single quotes ' or double dollar signs \$\$.
+- **What are the output fields of directory table result?**
+Relative_path and Modified_date
```

Comment:
the full list currently is relative_path, size, last_modified, md5, file_url, etag. But guessing your exam q had only those 2 main ones. (I didnt get this question )